src/ucs/sys/topo.c

```diff
@@ -19,6 +19,12 @@ ucs_sys_bus_id_t ucs_sys_bus_id_unknown = { .domain   = 0xffff,
                                             .function = 0xff
                                           };
 
+ucs_sys_bus_id_t ucs_sys_bus_id_local = { .domain   = 0xffff,
+                                          .bus      = 0xff,
+                                          .slot     = 0xff,
+                                          .function = 0xfe
+                                        };
```

Comment:
why needed?

Reply:
@yosefe  This is what we decided to return when cuda or sm transports are queried. This will UCP distinguish system device as different from unknown but the transport doesn't have to return the specific device id (such as gpu0, gpu1, ...) as part of lane to system device index translation. When lane -> device index translates to bus_id_local then distance query functions can prioritize that lane over others if it can support the specific memory type. An example is a follows:

```
 ucs_status_t
 uct_cuda_base_query_devices(uct_md_h md, uct_tl_device_resource_t **tl_devices_p,
                            unsigned *num_tl_devices_p)
 {
+    ucs_status_t status;
+    ucs_sys_device_t device_index;
+
+    status = ucs_topo_find_device_by_bus_id(&ucs_sys_bus_id_local,
+                                            &device_index);
+    if (status != UCS_OK) return status;
+
     return uct_single_device_resource(md, UCT_CUDA_DEV_NAME, UCT_DEVICE_TYPE_ACC,
+                                      device_index,
                                       tl_devices_p, num_tl_devices_p);
 }
```

Reply:
1. this is not needed in current PR
2. no need to define bus_id struct, just a special value for ucs_sys_device_t

Reply:
Removed local device.